druid-shell/src/windows/mod.rs

```diff
@@ -878,8 +881,13 @@ unsafe fn choose_adapter(factory: *mut IDXGIFactory2) -> *mut IDXGIAdapter {
         if !SUCCEEDED((*factory).EnumAdapters(i, &mut adapter)) {
             break;
         }
-        let mut desc: DXGI_ADAPTER_DESC = mem::uninitialized();
-        (*adapter).GetDesc(&mut desc);
+        let mut desc = mem::MaybeUninit::uninit();
+        let hr = (*adapter).GetDesc(desc.as_mut_ptr());
+        if !SUCCEEDED(hr) {
+            error!("Failed to get adapter description: {:?}", Error::Hr(hr));
+            break;
+        }
+        let mut desc: DXGI_ADAPTER_DESC = desc.assume_init();
```

Comment:
It feels like this doesn't really need to be `mut`. The problem is that our "util" module takes a `LPWSTR`, which is more idiomatically C. I think the solution is to use the [from_wide_ptr_null()](https://docs.rs/wio/0.2.2/i686-pc-windows-msvc/wio/wide/trait.FromWide.html#method.from_wide_ptr_null) method from wio and remove our hand-rolled "util" stuff, but that's arguably a separate issue from the scope of this PR, which is specifically about uninit.

Reply:
Sounds good. I'll add an issue for it.